src/Core/Infrastructure/Configuration/MetaService/Repository/DbMetaServiceFactory.php

```diff
@@ -35,5 +35,62 @@ class DbMetaServiceFactory
      */
     public static function createFromRecord(array $data): MetaService
     {
+        return (new MetaService(
+            (int) $data['id'],
+            $data['name'],
+            self::normalizeCalculationType($data['calculation_type']),
+            (int) $data['meta_selection_mode'],
+            self::normalizeDataSourceType((int) $data['data_source_type'])
+        ))
+        ->setWarning(self::getIntOrNull($data['warning']))
+        ->setCritical(self::getIntOrNull($data['critical']))
+        ->setOutput($data['output'])
+        ->setMetric($data['metric'])
+        ->setActivated((int) $data['is_activated'] === 1)
+        ->setRegexpSearchServices($data['regexp_search_services']);
+    }
+
+    /**
+     * This function will normalize the calculation type coming from the database
+     *
+     * @param string|null $calculationType
+     * @return string
+     */
+    private static function normalizeCalculationType(?string $calculationType): string
+    {
+        switch ($calculationType) {
+            case 'AVE':
+                return MetaService::CALCULTATION_TYPE_AVERAGE;
+            case 'MIN':
+                return MetaService::CALCULTATION_TYPE_MINIMUM;
+            case 'MAX':
+                return MetaService::CALCULTATION_TYPE_MAXIMUM;
+            case 'SOM':
+                return MetaService::CALCULTATION_TYPE_SUM;
+            default:
+                return MetaService::CALCULTATION_TYPE_AVERAGE;
+        }
```

Comment:
juste for info (PHP8) ;)
```suggestion
        return match ($calculationType) {
            'AVE' => MetaService::CALCULTATION_TYPE_AVERAGE,
            'MIN' => MetaService::CALCULTATION_TYPE_MINIMUM,
            'MAX' => MetaService::CALCULTATION_TYPE_MAXIMUM,
            'SOM' => MetaService::CALCULTATION_TYPE_SUM,
            default => MetaService::CALCULTATION_TYPE_AVERAGE
        }
```